ui/src/services/api.js

```diff
@@ -0,0 +1,53 @@
+import axios from 'axios';
+import { Config, Core_v1Api } from '@kubernetes/client-node';
+
+const api = axios.create({
+  baseURL: 'https://localhost:8080/api/v1',
+  timeout: 1000
+});
+
+//Basic Auth
+export const authenticate = ({ username, password }) => {
+  localStorage.removeItem('token');
+  const token = btoa(username + ':' + password); //base64Encode
+  return api
+    .get('/', {
+      headers: {
+        Authorization: 'Basic ' + token
+      }
+    })
+    .then(() => {
+      localStorage.setItem('token', token);
+      return {
+        response: {
+          username,
+          password,
+          token
+        }
+      };
+    })
+    .catch(errors => ({ errors: errors.response.data }));
+};
+
+export const logout = () => {
+  localStorage.clear();
+};
+
+export const getNodes = ({ token }) => {
+  const config = new Config('https://localhost:8080', token, 'Basic');
+  const coreV1 = config.makeApiClient(Core_v1Api);
+
+  return coreV1
+    .listNode()
+    .then(nodesResponse => {
+      return nodesResponse.body.items.map(node => ({
```

Comment:
Hmh... Wouldn't it be better if the code here simply returns plain `Node` objects, and let the renderer code pick whatever fields it wants?

Say we want to display `node.spec.podCIDR` in some widget. Then the code change would require a change in the widget (of course), but also here, which seems odd.

Reply:
I see but I am not fan to have lots of "useless" data in the input of the components.

Reply:
Then do the mangling somewhere else. This is an API accessing module, so it should return API objects as-is, IMHO.

Also, you're not really giving 'useless' input: your component gets (a list of) `Node` objects, that's it. Which parts they consume from such object is up to the component.

Reply:
OK to move it else where.
The thing is we store the Node objects in the Redux store, I think it is better to keep it as small as possible